.github/workflows/dep_code_checks.yml

```diff
@@ -0,0 +1,107 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+
+name: Code Checks
+
+on:
+  workflow_call:
+    inputs:
+      docs_only:
+        description: Skip building if docs only
+        required: false
+        type: string
+        default: "false"
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: full
+
+permissions:
+  id-token: write
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  linux-checks:
+    if: ${{ inputs.docs_only == 'false' }}
+    timeout-minutes: 30
+    runs-on: ["self-hosted", "Linux", "X64", "1ES.Pool=hld-kvm-amd"]
+    steps:
+      - uses: actions/checkout@v6
+
+      - uses: hyperlight-dev/ci-setup-workflow@v1.8.0
+        with:
+          rust-toolchain: "1.89" 
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Fix cargo home permissions
+        run: |
+          sudo chown -R $(id -u):$(id -g) /opt/cargo || true
+          sudo chmod -R u+rwX /opt/cargo || true
+
+      - name: Rust cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          shared-key: "code-checks-linux"
+          cache-on-failure: "true"
```

Comment:
This does not seem to cache guests, which below `just clippy-exhaustive` could benefit from, see https://github.com/hyperlight-dev/hyperlight/actions/runs/20116574753/job/57731955907#step:8:232. We should probably move `just clippy-guests` out of `just clippy-exhaustive`, it could be a source of confusion

Reply:
I fixed the caching